sys/kern/signal.c

```diff
@@ -397,6 +397,34 @@ void sig_pgkill(pgrp_t *pg, ksiginfo_t *ksi) {
   }
 }
 
+void sig_onexec(proc_t *p) {
+  assert(mtx_owned(&p->p_lock));
+  thread_t *td = p->p_thread;
+
+  /* The signal mask, pending and ignored signals remain unchanged.
+   * Caught signals have their action reset to SIG_DFL.
+   * If a pending signal becomes ignored due to its default action,
+   * it is discarded. */
+  for (signo_t sig = 1; sig < NSIG; sig++) {
+    if (sig_ignored(p->p_sigactions, sig)) {
+      /* Invariant check. */
```

Comment:
It's clear that next line is invariant check, while it may be not obvious why you check it. Please improve the comment.

Reply:
I don't _need_ to check it, but it's a good opportunity to do so. I have changed the comment to say exactly what the invariant is.